samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/MainActivity.kt

```diff
@@ -0,0 +1,162 @@
+
+    private lateinit var viewMvc: MainActivityViewMvc
+
+    private lateinit var trackingNode: TrackingNode
+    private lateinit var imageNodeHelper: ImageNodeHelper
+    private lateinit var viewNodeHelper: ViewNodeHelper
+    private lateinit var nodeRotationHelper: NodeRotationHelper
+
+    private val nodeList: MutableList<AnchorNode> = mutableListOf()
+
+    private var nodeType = NodeType.VIEW_NODE
+
+    ////////////////////
+    // region Lifecycle Methods
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewMvc = MainActivityViewMvcImpl(
+            this,
+            layoutInflater,
+            null
+        )
+        setContentView(viewMvc.getView().root)
+        viewMvc.registerListener(this)
+
+        viewMvc.getSceneView().apply {
+            initObjects(this)
+            configureArSession(this)
+            handleOnSessionUpdated(this)
+
+            lifecycle = this@MainActivity.lifecycle
+        }
+    }
+
+    private fun initObjects(arSceneView: ARSceneView) {
+        trackingNode = TrackingNode(
+            this@MainActivity,
+            arSceneView
+        )
+        imageNodeHelper = ImageNodeHelper(
+            this@MainActivity,
+            arSceneView,
+            this@MainActivity::onEvent
+        )
+        viewNodeHelper = ViewNodeHelper(
+            this@MainActivity,
+            arSceneView,
+            this@MainActivity::onEvent
+        )
+        nodeRotationHelper = NodeRotationHelper()
+    }
+
+    private fun configureArSession(arSceneView: ARSceneView) {
+        arSceneView.configureSession { session, config ->
+            config.depthMode = when (session.isDepthModeSupported(Config.DepthMode.AUTOMATIC)) {
+                true -> Config.DepthMode.AUTOMATIC
+                else -> Config.DepthMode.DISABLED
+            }
+            config.instantPlacementMode = Config.InstantPlacementMode.DISABLED
+            config.lightEstimationMode = Config.LightEstimationMode.ENVIRONMENTAL_HDR
+            config.planeFindingMode = Config.PlaneFindingMode.DISABLED
+        }
+    }
+
+    private fun handleOnSessionUpdated(arSceneView: ARSceneView) {
+        arSceneView.onSessionUpdated = { _, frame ->
+            trackingNode.frameUpdate(frame)
+            nodeRotationHelper
+                .updateRotationOfViewNodes(
+                    viewMvc.getSceneView(),
+                    nodeList
+                )
+            viewMvc.updateBtnEnabledState(trackingNode.isTracking())
+        }
+    }
+
+    override fun onDestroy() {
+        viewMvc.unregisterListener(this)
+        super.onDestroy()
+    }
+    // endregion Lifecycle Methods
+    ////////////////////
+
+
+    ///////////////////////////////
+    // region MainActivityViewMvc.Listener
+    override fun updateActiveNodeType(nodeType: NodeType) {
+        this.nodeType = nodeType
+    }
+
+    override fun handleOnAddClicked() {
```

Comment:
I'd rather call this method `onAddClick` or `handleAddClick`.

Reply:
I can go with that